### PR TITLE
CI: use conda-forge on 3.6 build

### PR DESCRIPTION
**Pull request by jreback**

### CODE REVIEW

**Comment by jreback** [comment]
xref https://github.com/pandas-dev/pandas/issues/15662 - build failures on scipy 0.19.0

File: pandas/tests/sparse/test_frame.py [hunk 1]
```diff
@@ -1132,7 +1132,7 @@ def test_isnotnull(self):
 @pytest.mark.parametrize('dtype', [object, bool, int, float, np.uint16])
```
Comment (kernc) [inline pandas/tests/sparse/test_frame.py]: You might as well push something to remove this object. :stuck_out_tongue:

Comment (jreback) [inline pandas/tests/sparse/test_frame.py]: should we just not test this at all (object I mean)? I don't actually have scipy 0.19.0 installed, so can't really test this.

Comment (kernc) [inline pandas/tests/sparse/test_frame.py]: For what I quickly tested (0.18.1) only simple methods and properties, like `.tocoo()`, `todok()`, `.dtype`, `.shape`, work with spmatrices of object dtype. While we do preserve this dtype if encountered, I see little harm in skipping it.

Comment (jreback) [inline pandas/tests/sparse/test_frame.py]: ok separated out the ``object`` test. skips on 0.19.0.

**Pull request closed without merging**